### PR TITLE
Fix internal development mode causing Apex extension issues

### PR DESCRIPTION
**Pull request by lcampos**
### What does this PR do?
Add missing api methods to the internal development mode which cause the Apex extension to fail during activation

### What issues does this PR fix or reference?
@W-6695536@, #1670 , #1678

### CODE REVIEW

File: packages/salesforcedx-vscode-core/package.json [hunk 1]
```diff
@@ -710,11 +710,6 @@
           "type": "boolean",
           "default": false,
           "description": "%push_or_deploy_on_save_enabled_description%"
-        },
```
Comment (lcampos) [inline packages/salesforcedx-vscode-core/package.json]: Customers are enabling this setting by mistake so I'm removing it from the Settings menu.

File: packages/salesforcedx-vscode-core/src/index.ts [hunk 2]
```diff
@@ -460,6 +460,16 @@ export async function activate(context: vscode.ExtensionContext) {
 
     // Api
     const internalApi: any = {
+      channelService,
```
Comment (lcampos) [inline packages/salesforcedx-vscode-core/src/index.ts]: Missing methods needed by the Apex extension, Apex Replay Debugger and Apex Debugger.

File: packages/salesforcedx-vscode-core/package.json (hunk 1 above)
Comment (brpowell) [reply packages/salesforcedx-vscode-core/package.json]: How is it enabled now?

Comment (lcampos) [reply packages/salesforcedx-vscode-core/package.json]: Like any of the other settings, via the settings.json file or the Settings UI.

Comment (brpowell) [reply packages/salesforcedx-vscode-core/package.json]: Gotcha, that's what I figured. Wasn't sure if it was "deregistered" or something by doing this.

**Pull request merged**